appIcons.js

```diff
@@ -166,7 +166,13 @@ const MyAppIcon = new Lang.Class({
 
     _updateRunningStyle: function() {
         this.parent();
```

Comment:
I'm not sure I understand:
1. Shouldn't dots be already hidden as no interesting windows are returned by ```getInterestingWindows```?
2. I totally don't understand the purpose of ```Shell.AppSystem.get_default().emit('installed-changed');```
3. I also don't understand the reason why focus should not be affected